### PR TITLE
improvement: improve screenshot-related links

### PR DESCRIPTION
**Pull request by WofWca**
## Proposed changes

See commit messages.

## Checklist

- [ ] Lint and unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added documentation to describe my feature.
- [x] I have squashed my commits into logic units.
- [x] I have described the changes in the commit messages.

## Other information

Did not test lol.

image

image

### CODE REVIEW

File: weblate/templates/screenshots/screenshot_show.html [hunk 1]
```diff
@@ -1 +1,2 @@
-<a href="{{ screenshot.image.url }}" class="thumbnail" alt="{{ screenshot.name }}" title="{{ screenshot.name }}" data-edit="{% url 'screenshot' pk=screenshot.pk %}"><img class="img-responsive" src="{{ screenshot.image.url }}" /></a>
+{% perm 'screenshot.edit' screenshot.translation.component as user_can_edit_screenshot %}
+<a href="{{ screenshot.image.url }}" class="thumbnail" alt="{{ screenshot.name }}" title="{{ screenshot.name }}" data-details-url="{% url 'screenshot' pk=screenshot.pk %}"{% if user_can_edit_screenshot %} data-can-edit{% endif %}><img class="img-responsive" src="{{ screenshot.image.url }}" /></a>
```
[the remark this answers is not in the document]
Comment (WofWca) [reply weblate/templates/screenshots/screenshot_show.html:2]: It was this way before. I think `alt` is meant to be moved to ``, not `<a>`.

File: weblate/templates/base.html [hunk 2]
```diff
@@ -340,7 +340,7 @@ <h4 class="modal-title" id="screenshotModal"></h4>
         <img src="" id="imagepreview" class="img-responsive">
       </div>
       <div class="modal-footer">
-        <a class="btn btn-warning" id="modalEditLink">{% trans "Edit" %}</a>
+        <a class="btn btn-warning" id="modalDetailsLink" data-edit-text="{% trans 'Edit' %}">{% trans "Screenshot details" %}</a>
```
Comment (nijel) [inline weblate/templates/base.html:343]:
```suggestion
        <a class="btn btn-warning" id="modalDetailsLink" data-edit-text="{% trans 'Edit screenshot' %}">{% trans "Screenshot details" %}</a>
```

File: weblate/templates/screenshots/screenshot_show.html (hunk 1 above)
Comment (nijel) [inline weblate/templates/screenshots/screenshot_show.html:2]: The permission is already checked in the parent template. While at it, we can also fix the `alt` attribute placement:

```suggestion
<a href="{{ screenshot.image.url }}" class="thumbnail" title="{{ screenshot.name }}" data-details-url="{% url 'screenshot' pk=screenshot.pk %}"{% if user_can_edit_screenshot %} data-can-edit{% endif %}><img class="img-responsive" src="{{ screenshot.image.url }}" alt="{{ screenshot.name }}" /></a>
```

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**